network/v1alpha1/custom.dnsnameresolver.testsuite.yaml

```diff
@@ -0,0 +1,313 @@
+    initial: |
+      apiVersion: network.openshift.io/v1alpha1
+      kind: DNSNameResolver
+      metadata:
+        name: example
+      spec:
+        name: www_example_com
+    expectedError: "DNSNameResolver.network.openshift.io \"example\" is invalid: spec.name: Invalid value: \"www_example_com\": spec.name in body should match '^(\\*\\.)?([A-Za-z0-9-]+\\.)*[A-Za-z0-9-]+\\.$'"
+  - name: Should not be able to specify DNS name without a trailing period
+    initial: |
+      apiVersion: network.openshift.io/v1alpha1
+      kind: DNSNameResolver
+      metadata:
+        name: example
+      spec:
+        name: www.example.com
+    expectedError: "DNSNameResolver.network.openshift.io \"example\" is invalid: spec.name: Invalid value: \"www.example.com\": spec.name in body should match '^(\\*\\.)?([A-Za-z0-9-]+\\.)*[A-Za-z0-9-]+\\.$'"
+  onUpdate:
+  - name: Should not update spec.name field
+    initial: |
+      apiVersion: network.openshift.io/v1alpha1
+      kind: DNSNameResolver
+      metadata:
+        name: example
+      spec:
+        name: www.example.com.
+    updated: |
+      apiVersion: network.openshift.io/v1alpha1
+      kind: DNSNameResolver
+      metadata:
+        name: example
+      spec:
+        name: www.newexample.com.
+    expectedError: "DNSNameResolver.network.openshift.io \"example\" is invalid: spec.name: Invalid value: \"string\": spec.name is immutable"
+  - name: Should be able to add valid IPv4 address
+    initial: |
+      apiVersion: network.openshift.io/v1alpha1
+      kind: DNSNameResolver
+      metadata:
+        name: example
+      spec:
+        name: www.example.com.
+    updated: |
+      apiVersion: network.openshift.io/v1alpha1
+      kind: DNSNameResolver
+      metadata:
+        name: example
+      spec:
+        name: www.example.com.
+      status:
+        resolvedNames:
+        - dnsName: www.example.com.
+          resolvedAddresses:
+          - ip: "192.168.1.1"
+            ttlSeconds: 60
+            lastLookupTime: "2023-08-08T15:07:04Z"
+    expected:  |
+      apiVersion: network.openshift.io/v1alpha1
+      kind: DNSNameResolver
+      metadata:
+        name: example
+      spec:
+        name: www.example.com.
+      status:
+        resolvedNames:
+        - dnsName: www.example.com.
+          resolvedAddresses:
+          - ip: "192.168.1.1"
+            ttlSeconds: 60
+            lastLookupTime: "2023-08-08T15:07:04Z"
+  - name: Should be able to add lowest valid IPv4 address
+    initial: |
+      apiVersion: network.openshift.io/v1alpha1
+      kind: DNSNameResolver
+      metadata:
+        name: example
+      spec:
+        name: www.example.com.
+    updated: |
+      apiVersion: network.openshift.io/v1alpha1
+      kind: DNSNameResolver
+      metadata:
+        name: example
+      spec:
+        name: www.example.com.
+      status:
+        resolvedNames:
+        - dnsName: www.example.com.
+          resolvedAddresses:
+          - ip: "0.0.0.0"
```

Comment:
This will still be valid when using type ipv4 for validation. As this field will store those IP addresses received on a DNS lookup from external nameservers, the field will not store any reserved addresses.